simapp/params/encoding.go

```diff
@@ -14,3 +16,14 @@ type EncodingConfig struct {
 	TxConfig          client.TxConfig
 	Amino             *codec.LegacyAmino
 }
+
+// RegisterCodecsTests registars codecs from the config to the global and
+// modules scope (mbm != nil).
+func (ec *EncodingConfig) RegisterCodecsTests(mbm module.BasicManager) {
+	std.RegisterLegacyAminoCodec(ec.Amino)
+	std.RegisterInterfaces(ec.InterfaceRegistry)
+	if mbm != nil {
+		mbm.RegisterLegacyAminoCodec(ec.Amino)
+		mbm.RegisterInterfaces(ec.InterfaceRegistry)
+	}
+}
```

Comment:
i don't think this is a good idea. simapp already calls these 2 functions.

In tests:
- if the test uses simapp -> nothing to do.
- if the test doesn't use simapp -> Use `MakeEncodingConfig`, and register manually all minimal amount of types.

Reply:
There were 2 different MakeEncodingConfig (or even 3 if we count the conditional build flags). Many x/../simulation tests go through the `/simapp` and relay on a global variable of ModuleBasic.